### DISCUSSION TITLE
[4] media manager modal height

### DISCUSSION THREAD

**Discussion by micker** [joomla/joomla-cms#32039] [General]
hello i don't understand why modal in media manager doesn't use full heigh have device
![image](https://user-images.githubusercontent.com/2539440/104696607-d636b400-570e-11eb-99f1-1828e857d7b7.png)

For now we lost many space and we can't display more than 1 line of images ... that not clean ...
i think we have 2 solutions :
- take modal full height (or 90%)
- display thumb smaller

ex 
![image](https://user-images.githubusercontent.com/2539440/104697087-873d4e80-570f-11eb-96cc-dfaa71f45ab4.png)

**Comment by dgrammatiko** [comment]
@micker it's an easy fix, change https://github.com/joomla/joomla-cms/blob/d49e4614f299b5f158f6a585de6f9c0313013168/layouts/joomla/form/field/media.php#L152

to `'bodyHeight'  => '80',` Where 80 means 80% of the viewport

**Comment by micker** [comment]
did you think it interesting to propose a pr ?

**Comment by dgrammatiko** [comment]
Go ahead

**Comment by infograf768** [comment]
@dgrammatiko 
That is the Image xtd, not the Change Image. The change has to be done in
https://github.com/joomla/joomla-cms/blob/0f955c88920d650e026b987732d49056d3e0be39/plugins/editors-xtd/image/image.php#L95-L102

Which does not  mean the other one is not also necessary ;)

**Reply by dgrammatiko** [reply]
Personally I would apply the fix to both those files for consistency

**Comment by infograf768** [comment]
Yep, that's why I wrote
> Which does not mean the other one is not also necessary ;)

Now, we have another issue: it is the waste of space between the images (`media-main`) and the footer.

**Comment by infograf768** [comment]
I mean
<img width="999" alt="Screen Shot 2021-01-17 at 18 23 25" src="https://user-images.githubusercontent.com/869724/104850739-28cfc600-58f1-11eb-93b9-94069ff975fc.png">

**Comment by micker** [comment]
yes need to change this too and find a solution for this
https://github.com/joomla/joomla-cms/issues/30880

**Comment by micker** [comment]
hi its realy strange
if i set height 90 that doesn't works ... but in width value works ...
![image](https://user-images.githubusercontent.com/2539440/104921939-5290f780-599a-11eb-9719-09d2a97a1592.png)
![image](https://user-images.githubusercontent.com/2539440/104921968-59b80580-599a-11eb-9209-00ccb4a7d4fd.png)
if i do 70 - 95
![image](https://user-images.githubusercontent.com/2539440/104922171-a0a5fb00-599a-11eb-8db8-1bcface74031.png)

![image](https://user-images.githubusercontent.com/2539440/104922073-7e13e200-599a-11eb-90c5-aa780f6be250.png)
width value is good

for margin bottom i can propose this
![image](https://user-images.githubusercontent.com/2539440/104921852-2d03ee00-599a-11eb-9ec8-079f02b99cc4.png)
if set

**Reply by infograf768** [reply]
`height: calc(100vh - 153px);` does not work here to reduce the gap between the footer and the body.

**Comment by micker** [comment]
yes maybe but at first (if we to test more) we need to find a work arround to have an 90% height for modal ,ot ?

**Comment by micker** [comment]
hello i update last ng with bs5
i can propose
`.jviewport-height70 {
    height: 80vh;
}`
=> strange we can't do a .jviewport-height80
result 
![image](https://user-images.githubusercontent.com/2539440/110756408-7100ca00-824a-11eb-895f-8e1c057f3e92.png)

```
.media-main {
    height: calc(100vh - 104px);
}
.media-browser {
    height: calc(100vh - 155px);
}
```
Result
![image](https://user-images.githubusercontent.com/2539440/110756634-bc1add00-824a-11eb-827a-f9fe776d440d.png)
Some return ?
@infograf768 @dgrammatiko @C-Lodder

**Reply by dgrammatiko** [reply]
@micker change https://github.com/joomla/joomla-cms/blob/0220b8b5e25dd7837acb2a994d644f915786c15c/layouts/libraries/html/bootstrap/modal/body.php#L38 to 

```php

if ($bodyHeight && $bodyHeight >= 20 && $bodyHeight <= 100)
```

**Reply by micker** [reply]
i didn't sse any difference, your code need to e added to my propose ?
![image](https://user-images.githubusercontent.com/2539440/110788290-ba631080-826e-11eb-9839-62ef419b9171.png)

**Comment by Equazar** [comment]
Hi, in Joomla 5.2 : add this lines in custom.css file

`.joomla-dialog-media-field > dialog{
  height: auto !important;
}`

It's do the trick

**Comment by Fedik** [comment]
In Joomla 5 the height is `80vh` for large screens, and `96vh` for smaller.
~~I would close this as not issue anymore.~~ Well, I just noticed it is Discussion :smile:
